### PR TITLE
Fixes 3292: Bump platform-go-middlewares to v2

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Checklist

- [ ] Tested with snapshotting feature disabled and pulp server URL not configured if appropriate

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-3292

File: pkg/middleware/enforce_identity.go [hunk 1]
```diff
@@ -66,10 +66,10 @@ func SkipAuth(c echo.Context) bool {
 
 func EnforceOrgId(next echo.HandlerFunc) echo.HandlerFunc {
 	return func(c echo.Context) error {
-		if identity.GetIdentityHeader(c.Request().Context()) == "" {
+		if identity.GetIdentityHeader(c.Request().Context()) == "" { // This is being deprecated
```
Comment (Andrewgdewar) [inline pkg/middleware/enforce_identity.go:69]: Not sure how to handle updating this block.

**Comment by jlsherrill** [comment]
overall looks good. i couldn't figure out how (if possible) to get service accounts in ephemeral, so once this is merged, we can test in stage

File: pkg/middleware/enforce_identity.go (hunk 1 above)
Comment (Andrewgdewar) [reply pkg/middleware/enforce_identity.go:69]: Thank you for the clarification! will update.

**Comment by Andrewgdewar** [comment]
> was there a mock here that needed to be updated? I see a bunch of file changes on the mock files, but it looks like they're all just updating the version in the header comment.

I believe there was one update in one file, I'll double check though, and remove all changes that aren't necessary.

**Comment by jlsherrill** [comment]
Can you try pushing this again to see if the snyk checks clear up?

**Comment by jlsherrill** [comment]
needs a rebase, that may clear up the sync checks

**Review by jlsherrill — APPROVED** [review]

**Comment by swadeley** [comment]
Hi, I deployed the backend image, API tests pass locally.

**Pull request merged**